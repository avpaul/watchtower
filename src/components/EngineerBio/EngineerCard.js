import React from 'react';
import PropTypes from 'prop-types';
import Loader from '../Loader/Loader';
import './EngineerBio.css';
import ProfileCard from './profileCard';
import ProjectCards from './ProjectCards';

const renderEngineerDetails = fellow => (
  <React.Fragment>
    <ProfileCard
      firstName={fellow.first_name}
      lastName={fellow.last_name}
      picture={fellow.picture}
    />
    <ProjectCards
      startDate={fellow.sims_start_date}
      endDate={fellow.sims_end_date}
      projectLevel="Simulations"
      picture={fellow.picture}
      projectName={fellow.sims_project}
      technologies={fellow.sims_project_technology}
    />
    <ProjectCards
      startDate={fellow.apprenticeship_start_date}
      projectName={fellow.apprenticeship_project}
      endDate={fellow.apprenticeship_end_date}
      picture={fellow.picture}
      projectLevel="Apprenticeship"
      technologies={fellow.apprenticeship_technology}
    />
  </React.Fragment>
);

const EngineerCard = ({ data }) => {
  const { fellow, loading } = data;
  return (
    <div>
      <h3 className="profile-card-header">My Profile</h3>
      <div className="bio-card text-center">
        {!loading && fellow ? renderEngineerDetails(fellow) : <Loader />}
      </div>
    </div>
  );
};

EngineerCard.propTypes = {
  data: PropTypes.shape().isRequired
};

export default EngineerCard;
