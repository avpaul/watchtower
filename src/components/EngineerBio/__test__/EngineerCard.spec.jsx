import React from 'react';
import { shallow } from 'enzyme';
import EngineerCard from '../EngineerCard';

describe('tests the Engineer card', () => {
  const props = {
    data: {
      fellow: {
        fellow_id: '-LJK1qECFdekucrwqqkh',
        manager_id: '-LGy4OuPDHCZCZvDuPz0',
        first_name: 'Cristian',
        last_name: 'Bartell',
        sims_start_date: '',
        sims_end_date: '',
        sims_project: 'AuthorsHaven',
        sims_project_technology: 'Python/Django',
        apprenticeship_start_date: '',
        apprenticeship_end_date: '',
        apprenticeship_project: 'WatchTower',
        apprenticeship_technology: 'PHP/JavaScript'
      },
      loading: false
    }
  };

  it('renders correctly when loading is false', () => {
    const wrapper = shallow(<EngineerCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when fellow data not passed [Return null view]', () => {
    const wrapper = shallow(<EngineerCard data={{ loading: true }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when loading is true', () => {
    const wrapper = shallow(
      <EngineerCard data={{ fellow: {}, loading: true }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
