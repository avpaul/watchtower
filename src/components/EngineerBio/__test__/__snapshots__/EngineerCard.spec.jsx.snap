// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the Engineer card renders correctly when fellow data not passed [Return null view] 1`] = `
<div
  className="engineer-bio-wrapper"
>
  <h3
    className="profile-card-header"
  >
    My Profile
  </h3>
  <div
    className="bio-card text-center"
  />
</div>
`;

exports[`tests the Engineer card renders correctly when loading is false 1`] = `
<div
  className="engineer-bio-wrapper"
>
  <h3
    className="profile-card-header"
  >
    My Profile
  </h3>
  <div
    className="bio-card text-center"
  >
    <EngineerDetails
      fellow={
        Object {
          "apprenticeship_end_date": "",
          "apprenticeship_project": "WatchTower",
          "apprenticeship_start_date": "",
          "apprenticeship_technology": "PHP/JavaScript",
          "fellow_id": "-LJK1qECFdekucrwqqkh",
          "first_name": "Cristian",
          "last_name": "Bartell",
          "manager_id": "-LGy4OuPDHCZCZvDuPz0",
          "sims_end_date": "",
          "sims_project": "AuthorsHaven",
          "sims_project_technology": "Python/Django",
          "sims_start_date": "",
        }
      }
      view=""
    />
  </div>
</div>
`;

exports[`tests the Engineer card renders correctly when loading is true 1`] = `
<div
  className="engineer-bio-wrapper"
>
  <h3
    className="profile-card-header"
  >
    My Profile
  </h3>
  <div
    className="bio-card text-center"
  >
    <EngineerDetails
      fellow={Object {}}
      view=""
    />
  </div>
</div>
`;
